### PR TITLE
Make credential entries respect the --password-value option

### PR DESCRIPTION
**Pull request by MaciejG604**
Should work like this - mentioned [here](https://scala-cli.virtuslab.org/docs/commands/publishing/publish-setup#sonatype-credentials)
Also documentation for repositories.credentials was wrong - it accepts a format with repository address at the beginning and an optional realm at the end.

### CODE REVIEW

File: modules/config/src/main/scala/scala/cli/config/Keys.scala [hunk 1]
```diff
@@ -341,7 +342,7 @@ object Keys {
 
   val publishCredentials: Key[List[PublishCredentials]] = new Key[List[PublishCredentials]] {
     override val description: String =
-      "Publishing credentials, syntax: s1.oss.sonatype.org value:user value:password"
```
Comment (MaciejG604) [inline modules/config/src/main/scala/scala/cli/config/Keys.scala:344]: Should this be kept as an example address as it was? WDYT?

Comment (Gedochao) [reply modules/config/src/main/scala/scala/cli/config/Keys.scala:344]: it's nice to have an example in a guide or general command doc, but IMO reference docs & help outputs (like here) should provide general syntax instead
tl;dr don't keep it here, but it may be worth having in a guide doc or smth

**Review by Gedochao — APPROVED** [review]
LGTM

File: modules/config/src/main/scala/scala/cli/config/Keys.scala (hunk 1 above)
Comment (MaciejG604) [reply modules/config/src/main/scala/scala/cli/config/Keys.scala:344]: There are examples of the publish credentials already in `publish` and `publish setup` docs.
